# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1521075507 25200
#      Wed Mar 14 17:58:27 2018 -0700
# Branch stable
# Node ID 92aaa85c06b8caa494aa5089cad1402d5889bb98
# Parent  bdecc746584096808d4cca1e0644dba996576a07
build: use normal automake syntax for installing Qt help (bug #53006)

* Makefile.am (octdoc_DATA): Initialize to empty.
(install-data-local): No longer depend on doc-interpreter-install-doc-local.

* doc/interpreter/module.mk (octdoc_DATA): Include $(OCTAVE_QTHELP_FILES).
(doc-interpreter-install-doc-local): Delete.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -92,16 +92,17 @@ TAR_OPTIONS = $(REPRODUCIBLE_TAR_FLAGS) 
 
 CONFIG_SUBDIRS = @subdirs@
 
 null =
 ldpreloadsep = ${null}@ldpreloadsep@${null}
 
 image_DATA =
 octdata_DATA =
+octdoc_DATA =
 octetc_DATA =
 octlocale_DATA =
 
 DOC_TARGETS =
 
 BUILT_DISTFILES =
 BUILT_NODISTFILES =
 EXTRA_DIST =
@@ -394,17 +395,17 @@ DIRS_TO_MAKE = \
   $(localveroctfiledir) \
   $(localarchlibdir) \
   $(localapiarchlibdir) \
   $(localverarchlibdir)
 
 installdirs-local:
 	$(MKDIR_P) $(addprefix $(DESTDIR), $(DIRS_TO_MAKE))
 
-install-data-local: installdirs-local doc-interpreter-install-doc-local
+install-data-local: installdirs-local
 
 clean-local: doc-clean
 
 distclean-local:
 	if [ "x${srcdir}" != "x." ]; then rm -f HG-ID; fi
 
 maintainer-clean-local: doc-maintainer-clean
 
diff --git a/doc/interpreter/module.mk b/doc/interpreter/module.mk
--- a/doc/interpreter/module.mk
+++ b/doc/interpreter/module.mk
@@ -189,16 +189,19 @@ OCTAVE_HTML_DIR = %reldir%/octave.html
 OCTAVE_HTML_TMP_DIR = $(OCTAVE_HTML_DIR:.html=.htp)
 OCTAVE_HTML_STAMP = $(OCTAVE_HTML_DIR)/.octave-html-stamp
 
 OCTAVE_CSS = %reldir%/octave.css
 HTMLDIR_CSS = $(OCTAVE_HTML_DIR)/octave.css
 
 OCTAVE_QTHELP_FILES = %reldir%/octave_interpreter.qhc %reldir%/octave_interpreter.qch
 
+octdoc_DATA += \
+  $(OCTAVE_QTHELP_FILES)
+
 $(srcdir)/%reldir%/octave.info: $(DOC_IMAGES_TXT) $(octave_TEXINFOS)
 %reldir%/octave.dvi: $(DOC_IMAGES_EPS) $(octave_TEXINFOS)
 %reldir%/octave.pdf: $(DOC_IMAGES_PDF) $(octave_TEXINFOS)
 $(OCTAVE_HTML_STAMP): $(DOC_IMAGES_PNG) $(octave_TEXINFOS)
 
 $(OCTAVE_QTHELP_FILES): $(OCTAVE_HTML_STAMP) $(srcdir)/%reldir%/prepare_qhelp.py
 	$(AM_V_GEN)rm -f $(OCTAVE_QTHELP_FILES) && \
 	$(PYTHON) $(srcdir)/%reldir%/prepare_qhelp.py %reldir%/octave_interpreter octave.html && \
@@ -296,20 +299,16 @@ doc_EXTRA_DIST += \
 doc-interpreter-dist-hook:
 	@$(GREP) '#define HAVE_COLAMD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing COLAMD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_CHOLMOD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing CHOLMOD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_UMFPACK 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing UMFPACK library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_QHULL 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing QHULL library.  Cannot package distribution!" ; exit 1; }
 
 $(MUNGED_TEXI_SRC): $(DOCSTRING_FILES)
 
-doc-interpreter-install-doc-local:
-	$(MKDIR_P) $(octdocdir)
-	cp $(OCTAVE_QTHELP_FILES) $(octdocdir)
-
 ## These two texi files have an additional dependency through the
 ## @EXAMPLEFILE macro.
 %reldir%/oop.texi: $(examples_code_SRC)
 %reldir%/external.texi: $(examples_code_SRC)
 
 %.texi : %.txi %reldir%/munge-texi.pl | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(PERL) $(srcdir)/%reldir%/munge-texi.pl $(top_srcdir) $(DOCSTRING_FILES) < $< > $@-t && \
